### PR TITLE
Add a more specific error message when providing an invalid layout to fn.crop

### PR DESCRIPTION
**Pull request by jantonguirao**
## Category:
**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:

Improving error message when providing an invalid layout to crop operator

### Affected modules and functionalities:

Crop

### Key points relevant for the review:

Error message update

### Tests
- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [x] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Review by JanuszL — APPROVED** [review]

File: dali/operators/image/crop/crop.h [hunk 1]
```diff
@@ -40,10 +40,21 @@ class Crop : public SliceBase<Backend> {
...
+        ImageLayoutInfo::HasChannel(in_layout) &&
+            (ImageLayoutInfo::IsImage(in_layout) || VideoLayoutInfo::IsVideo(in_layout)),
+        make_string("Unexpected layout: ", in_layout, "\n\n",
+                    "Expected a non-empty layout including spatial dimensions "
+                    "(H, W, D) and a channel dimension. Examples: \"HWC\", \"DHWC\", \"FHWC\".\n\n"
+                    "To specify a layout manually, use ``fn.reshape``.\n"
+                    "Example:\n\n"
+                    "  ``data = fn.reshape(data, layout=\"HWC\")"));
+
+    DALI_ENFORCE(
```
Comment (stiepan) [inline dali/operators/image/crop/crop.h:54]: Nitpick: This is checked anyway by the Operator<Backend> (although in Run method so maybe `crop` needs that assured earlier?).

**Review by stiepan — APPROVED** [review]

File: dali/operators/image/crop/crop.h (hunk 1 above)
Comment (jantonguirao) [reply dali/operators/image/crop/crop.h:54]: There's another PR that actually adds layout info to this op. I think I will merge that one first, then update this one.

Comment (stiepan) [reply dali/operators/image/crop/crop.h:54]: Sure.

**Comment by jantonguirao** [comment]
Redundant after https://github.com/NVIDIA/DALI/pull/3722

**Pull request closed without merging**